CI: Add codespell config and fix typos

yarikoptic opened the pull request:
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
[DATALAD RUNCMD] Do interactive fixing of some ambigous typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 2",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

+ fixed few manually and melded here in frontend/src/css/vendor/icons/material-design-icons.css which was ignored

 Conflicts upon rebased -- files were removed in develop:
	frontend/src/css/vendor/icons/fonts/MaterialIcons-Regular.json
	frontend/src/css/vendor/icons/material-design-icons.css

yarikoptic committed:
A few more skips for codespell

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/photoprism/photoprism?pullRequest=4506) <br/>All committers have signed the CLA.

yarikoptic commented:
is there a way to trigger some testing?  otherwise -- this is an easy to review PR, don't get scared by number of files, quite often it is just one typo fix per file.

lastzero commented:
@yarikoptic I apologize for not getting back to you any sooner!

The reason we've been so reluctant with this is because of security concerns regarding GitHub Actions (especially when adding/modifying them through pull requests). See, for example:

- https://medium.com/@simardeep.oberoi/github-actions-supply-chain-attack-how-tj-actions-changed-files-leaked-secrets-and-what-you-need-9e2677e07f68

That being said, we'd be happy to merge the spelling fixes without the action configuration and/or add an additional Make target that [runs a spell check](https://github.com/golangci/golangci-lint/blob/f15e6c4865d5b55c04e192a6ac49c4117ff0b1c7/pkg/golinters/misspell/misspell.go) locally, just like eslint. It would be worth looking at https://github.com/golangci/golangci-lint for this as well, since `golangci-lint` might be a better and more versatile choice for our needs.

Would that be possible? I'll be happy to help with this and answer any questions :)

yarikoptic commented:
sure -- concerns are well understood I must say especially if you have/use some secrets (I might have missed but I do not think you do yet). Anything is fine with me. I can easily redo and drop CI action.  Just let me know if you would have time now to review so we do not breed conflicts and require re-doing?

yarikoptic committed:
Add a few codespell:ignore pragmas on intentional typos like uses

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout fixing few left typos automagically

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic commented:
actually, "easier done than said" so here is an updated -- dropped CI, rebased, added a few ignore pragmas, less changes to review

lastzero commented:
Excellent, thank you very much! I'll go ahead and merge this.

lastzero commented:
@yarikoptic This is now included in our [preview build](https://docs.photoprism.app/release-notes/#development-preview)! Note that I had to fix some minor issues, for example using `#` for comments is not allowed in Go. I assume that using `//` instead shouldn't affect the codespell annotation.

yarikoptic commented:
Oh, so sorry that my Python rotten mind saw everything as Python code. Yes, it should work with a proper to the language comment
